File: src/main/java/com/procurement/submission/model/dto/OrganizationReference.java
package com.procurement.submission.model.dto;

import java.util.List;
import lombok.Data;

@Data
public class OrganizationReference {
    public String name;
    public Integer id;
    public Identifier identifier;
    public Address address;
    public List<Identifier> additionalIdentifiers;
    public ContactPoint contactPoint;
}

File: src/main/java/com/procurement/submission/model/dto/ocds/OrganizationReference.kt
package com.procurement.submission.model.dto.ocds

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import javax.validation.Valid
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

@JsonInclude(JsonInclude.Include.NON_NULL)
data class OrganizationReference(

        @JsonProperty("id")
        var id: String?,

        @JsonProperty("name") @Size(min = 1) @NotNull
        val name: String,

        @JsonProperty("identifier") @Valid @NotNull
        val identifier: Identifier,

        @JsonProperty("address") @Valid @NotNull
        val address: Address,

        @JsonProperty("additionalIdentifiers") @Valid
        val additionalIdentifiers: Set<Identifier>?,

        @JsonProperty("contactPoint") @Valid @NotNull
        val contactPoint: ContactPoint
)
